fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/TempTableScanPlan.java

```diff
@@ -70,14 +71,15 @@ public TempTableScanPlan(@Nonnull final Value tempTableReferenceValue) {
         this.tempTableReferenceValue = tempTableReferenceValue;
     }
 
+    @SuppressWarnings("unchecked")
     @Nonnull
     @Override
     public <M extends Message> RecordCursor<QueryResult> executePlan(@Nonnull final FDBRecordStoreBase<M> store,
                                                                      @Nonnull final EvaluationContext context,
                                                                      @Nullable final byte[] continuation,
                                                                      @Nonnull final ExecuteProperties executeProperties) {
-        final var tempTable = (TempTable)this.tempTableReferenceValue.eval(store, context);
-        return tempTable.getReadCursor(continuation);
+        final var tempTable = Objects.requireNonNull((TempTable)this.tempTableReferenceValue.eval(store, context));
+        return new IteratorCursor<>(store.getExecutor(), tempTable.getIterator());
```

Comment:
Don't you have to apply the continuation here when you create the `IteratorCursor`?

Reply:
Added a method that returns the underlying synchronized `List` itself, and returning a `ListCursor` on top which has proper continuation semantics.

I might add tests around scanning with continuations in a subsequent PR.